Dalamud/Interface/Internal/Windows/PluginImageCache.cs

```diff
@@ -234,14 +287,43 @@ private void FrameworkOnUpdate(Framework framework)
 
         private async void DownloadTask()
         {
-            while (!this.downloadToken.Token.IsCancellationRequested)
+            var token = this.downloadToken.Token;
+            var runningTasks = new List<Task>();
+            var pendingFuncs = new List<Tuple<ulong, Func<Task>>>();
+            while (true)
```

Comment:
Maybe we want to sleep in this loop, to not use up CPU for something insignificant - last iteration didn't have one either though

Reply:
1. If there aren't any further pending tasks, TryTake will block.
2. If there are any, then it will block when there already are 8 tasks running.

This loop shouldn't be hogging CPU times, so sleep won't be necessary, but probably can limit concurrent decoding to the number of cores available, while having separate limit of 8 for image download.